### ISSUE TITLE
Android build error Expo 51 - `Plugin ... was not found in any of the following sources`

### ISSUE THREAD

**Issue by robwalkerco** [DataDog/expo-datadog#33]
**Describe what happened**
After upgrading to Expo 51, it's not possible to build the Android app.
It appears this may be related to the version of Gradle being updated to 8.6

**Steps to reproduce the issue:**

 - Upgrade an app that uses `expo-datadog` from Expo 50 to Expo 51
 - Build the Android app
 - Get build error (below)

```
FAILURE: Build failed with an exception.

* Where:
Build file '<REDACTED>/android/app/build.gradle' line: 2

* What went wrong:
Plugin [id: 'com.datadoghq.dd-sdk-android-gradle-plugin', version: '1.+'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'com.datadoghq.dd-sdk-android-gradle-plugin:com.datadoghq.dd-sdk-android-gradle-plugin.gradle.plugin:1.+')
  Searched in the following repositories:
    Gradle Central Plugin Repository
```

I've resolved this issue by setting an explicit `datadogGradlePluginVersion` version of `1.13.0`, which allows the build to complete, but it would be great if that was not required.

Example plugin config to get the build working again
```js
expo: {
  ...
  plugins: [
    ...
    [
        'expo-datadog',
        {
          errorTracking: {
            datadogGradlePluginVersion: '1.13.0',
          },
        },
      ],
  ]
}
```

**Describe what you expected:**
The build should complete

**Additional context**

```
"@datadog/mobile-react-native": "^2.3.2",
"@datadog/mobile-react-navigation": "^2.3.2",
"expo": "^51.0.1",
"expo-datadog": "^50.2.0",
"react-native": "0.74.1",
...
```

**Comment by marco-saia-datadog** [comment]
Hey @robwalkerco,

The update to Expo 51 is in our roadmap, and we will grant support soon with `datadog-expo@51.0.0`.

I will keep you posted here as soon as we have an open PR.

Thank you for your valuable feedback!

**Comment by raghavsethi** [comment]
Subscribing - we'd like to switch our company to Datadog and this is blocking us at the moment.

**Comment by panasenco** [comment]
I work at a company that's a new Datadog customer and we are also blocked by this. Is there a timeline that you can share for supporting expo 51?

**Comment by marco-saia-datadog** [comment]
@marcato15

The delay was mainly caused by the fact that one of our engineers, responsible for the Expo updates, recently left the company. There has been a period of knowledge transfer and onboarding for the new engineer, and an unusual amount of critical issues to solve in our React Native SDK.

Future updates will be quicker, we apologise for the inconvenience 🙏

**Comment by marco-saia-datadog** [comment]
@robwalkerco @raghavsethi @panasenco @marcato15 

Sorry for the long wait!

We have released [expo-datadog@51.0.0](https://www.npmjs.com/package/expo-datadog/v/51.0.0), which solves the build issue on Android and grants support for Expo 51.

**Comment by marcato15** [comment]
> @robwalkerco @raghavsethi @panasenco @marcato15
> 
> Sorry for the long wait!
> 
> We have released [expo-datadog@51.0.0](https://www.npmjs.com/package/expo-datadog/v/51.0.0), which solves the build issue on Android and grants support for Expo 51.

Thank you for the explanation. That is very helpful.

**Comment by raghavsethi** [comment]
Thank you, much appreciated!
